code/game/objects/structures/beds_chairs/sofa.dm

```diff
@@ -1,67 +1,69 @@
-/obj/structure/chair/sofa
-	name = "old ratty sofa"
-	icon_state = "sofamiddle"
-	icon = 'icons/obj/sofa.dmi'
+/obj/structure/chair/fancy/sofa //like it's brother fancy chair, this is the father of all sofas
+	name = "old father sofa"
+	desc = "Now extint, this kind of sofa shouldn't even exists anymore, if you see this wild specimen, contact your local Nanotransen Anti-couch surfer department."
+	icon_state = "sofa_middle"
+	icon = 'icons/obj/beds_chairs/sofa.dmi'
 	buildstackamount = 1
 	item_chair = null
-	var/mutable_appearance/armrest
 
-/obj/structure/chair/sofa/Initialize(mapload)
-	armrest = mutable_appearance(icon, "[icon_state]_armrest", ABOVE_MOB_LAYER)
-	return ..()
-/obj/structure/chair/sofa/post_buckle_mob(mob/living/M)
-	. = ..()
-	update_armrest()
+/obj/structure/chair/fancy/sofa/old
+	color = rgb(141,70,0)
+	name = "old sofa"
+	desc = "An old design, but still does the job of being a sofa."
+	icon_state = "sofa_middle"
+	colorable = TRUE
 
-/obj/structure/chair/sofa/proc/update_armrest()
-	if(has_buckled_mobs())
-		add_overlay(armrest)
-	else
-		cut_overlay(armrest)
+/obj/structure/chair/fancy/sofa/old/left
+	icon_state = "sofa_end_left"
 
-/obj/structure/chair/sofa/post_unbuckle_mob()
-	. = ..()
-	update_armrest()
+/obj/structure/chair/fancy/sofa/old/right
+	icon_state = "sofa_end_right"
 
-/obj/structure/chair/sofa/corner/handle_layer() //only the armrest/back of this chair should cover the mob.
+/obj/structure/chair/fancy/sofa/old/corner
+	name = "impossible old sofa corner"
+	desc = "this kind of sofa shouldn't even exists at all, if you see this non-euclidan specimen, contact your local Nanotransen Anti-couch surfer department."
```

Comment:
This kind of sofa shouldn't even exist at all. If you see this non-euclidean specimen, contact your station's Anti-Couch Surfing Department.